src/xdp/rx.c

```diff
@@ -1082,6 +1086,12 @@ XdpRxQueueSetProgram(
         SwapParams.RxQueue = RxQueue;
         SwapParams.NewProgram = Program;
         XdpRxQueueSync(RxQueue, XdpRxQueueSwapProgram, &SwapParams);
+        if (OldCompiledProgram) {
+            //
+            // Free the old compiled program because it's no longer being used.
+            //
+            ExFreePoolWithTag(OldCompiledProgram, XDP_POOLTAG_PROGRAM);
```

Comment:
Forgot to free the old program when (Program == NULL and RxQueue->Program != NULL)